src/function/fetch.rs

```diff
@@ -272,4 +213,76 @@ where
 
         Some(memo)
     }
+
+    #[cold]
```

Comment:
I'd also add `#[inline(never)]`, because LLVM tends to be aggressive with inlining even with `#[cold]`.